### PR TITLE
Re-add actor count and revert recent AlwaysReplicate change

### PR DESCRIPTION
**Pull request by m-samiec**
Reverting the rep graph changes introduced here (https://github.com/spatialos/UnrealGDKTestGyms/pull/450) and added actor count logging.

### CODE REVIEW

**Review by mironec — APPROVED** [review]
Probably good, but not sure what the uasset changes are doing

File: Game/Source/GDKTestGyms/Private/CounterComponent.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+		UpdateCachedActorCounts();
+	}
+}
+
+int32 UCounterComponent::GetActorClassCount(TSubclassOf<AActor> ActorClass) const
+{
+	const int32* CachedCount = CachedClassCounts.Find(ActorClass);
+	if (CachedCount == nullptr)
+	{
+		UE_LOG(LogTemp, Warning, TEXT("Could not get actor class count for class %s"), *ActorClass->GetName());
```
Comment (mironec) [inline Game/Source/GDKTestGyms/Private/CounterComponent.cpp:33]: LogCounterComponent?

File: Game/Source/GDKTestGyms/Private/CounterComponent.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+	Timer -= DeltaTime;
+	if (Timer <= 0.0f)
+	{
+		Timer = ReportFrequency;
+		UpdateCachedActorCounts();
+	}
+}
+
+int32 UCounterComponent::GetActorClassCount(TSubclassOf<AActor> ActorClass) const
```
Comment (mironec) [inline Game/Source/GDKTestGyms/Private/CounterComponent.cpp:28]: is this function even used? wouldn't we just report the current situation in update? as in - do we need the caching?

Comment (m-samiec) [reply Game/Source/GDKTestGyms/Private/CounterComponent.cpp:28]: You're right this isn't currently used, but this is just reverting files that already existed, so will leave for the time being.

**Comment by m-samiec** [comment]
> Probably good, but not sure what the uasset changes are doing

Sorry should have specified - adding the Count Component back to the Game States, and outputting the counts of the actors we care about.

**Pull request merged**